recipes-security/optee-imx/optee-os_3.2.0.imx.bb

```diff
@@ -0,0 +1,94 @@
+
+SUMMARY = "OPTEE OS"
+DESCRIPTION = "OPTEE OS"
+HOMEPAGE = "http://www.optee.org/"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=69663ab153298557a59c67a60a743e5b"
+
+inherit deploy pythonnative autotools
+DEPENDS = "python-pycrypto-native u-boot-mkimage-native"
+
+SRCBRANCH = "imx_4.14.78_1.0.0_ga"
+OPTEE_OS_SRC ?= "git://source.codeaurora.org/external/imx/imx-optee-os.git;protocol=https"
+SRC_URI = "${OPTEE_OS_SRC};branch=${SRCBRANCH}"
+SRCREV = "6a52487eb0ff664e4ebbd48497f0d3322844d51d"
+
+S = "${WORKDIR}/git"
+B = "${WORKDIR}/build.${PLATFORM_FLAVOR}"
+
+# The platform flavor corresponds to the Yocto machine without the leading 'i'.
+PLATFORM_FLAVOR                 = "${@d.getVar('MACHINE')[1:]}"
+PLATFORM_FLAVOR_imx6qpdlsolox   = "mx6qsabresd"
+PLATFORM_FLAVOR_imx6ul7d        = "mx6ulevk"
+PLATFORM_FLAVOR_imx6ull14x14evk = "mx6ullevk"
+PLATFORM_FLAVOR_imx6ull9x9evk   = "mx6ullevk"
+PLATFORM_FLAVOR_imx6ulz14x14evk = "mx6ullevk"
+PLATFORM_FLAVOR_mx8mm   = "mx8mmevk"
+
+OPTEE_ARCH ?= "arm32"
+OPTEE_ARCH_armv7a = "arm32"
+OPTEE_ARCH_aarch64 = "arm64"
+
+# Optee-os can be built for 32 bits and 64 bits at the same time
+# as long as the compilers are correctly defined.
+# For 64bits, CROSS_COMPILE64 must be set
+# When defining CROSS_COMPILE and CROSS_COMPILE64, we assure that
+# any 32 or 64 bits builds will pass
+EXTRA_OEMAKE = "PLATFORM=imx PLATFORM_FLAVOR=${PLATFORM_FLAVOR} \
+                CROSS_COMPILE=${HOST_PREFIX} \
+                CROSS_COMPILE64=${HOST_PREFIX} \
+                NOWERROR=1 \
+                LDFLAGS= \
+		O=${B} \
+                CFG_SECURE_DATA_PATH=y \
+                CFG_TEE_SDP_MEM_BASE=0xCC000000 \
+                CFG_TEE_SDP_MEM_SIZE=0x02000000 \
+                CFG_TEE_SDP_NONCACHE=y \
+                "
+
+
+do_compile () {
+    unset LDFLAGS
+    export CFLAGS="${CFLAGS} --sysroot=${STAGING_DIR_HOST}"
+    oe_runmake -C ${S} all CFG_TEE_TA_LOG_LEVEL=0
+}
+
+
+do_deploy () {
+    install -d ${DEPLOYDIR}
+    ${TARGET_PREFIX}objcopy -O binary ${B}/core/tee.elf ${DEPLOYDIR}/tee.${PLATFORM_FLAVOR}.bin
+
+    if [ "${OPTEE_ARCH}" != "arm64" ]; then
+        IMX_LOAD_ADDR=`cat ${B}/core/tee-init_load_addr.txt` && \
+        uboot-mkimage -A arm -O linux -C none -a ${IMX_LOAD_ADDR} -e ${IMX_LOAD_ADDR} \
+            -d ${DEPLOYDIR}/tee.${PLATFORM_FLAVOR}.bin ${DEPLOYDIR}/uTee-${OPTEE_BIN_EXT}
+    fi
+
+    cd ${DEPLOYDIR}
+    ln -sf tee.${PLATFORM_FLAVOR}.bin tee.bin
+    cd -
+}
+
+do_install () {
+    install -d ${D}/lib/firmware/
+    install -m 644 ${B}/core/*.bin ${D}/lib/firmware/
+
+    # Install the TA devkit
+    install -d ${D}/usr/include/optee/export-user_ta_${OPTEE_ARCH}/
+
+    for f in ${B}/export-ta_${OPTEE_ARCH}/*; do
+        cp -aR $f ${D}/usr/include/optee/export-user_ta_${OPTEE_ARCH}/
+    done
+}
+
+addtask deploy after do_compile before do_install
+
+
+FILES_${PN} = "/lib/firmware/"
+FILES_${PN}-dev = "/usr/include/optee"
+INSANE_SKIP_${PN}-dev = "staticdev"
```

Comment:
Why?

Reply:
Without disabling staticdev I get the following QA warning. Again it appears copied from the Linaro optee recipe and has been present since the first git version [1].

[1] https://git.linaro.org/openembedded/meta-linaro.git/commit/meta-optee/recipes-security/optee/optee-os_git.bb?id=a5d6db26e1bf6a398fcaf87d5a8f2365cbc54090

WARNING: optee-os-3.2.0.imx-r0 do_package_qa: QA Issue: non -staticdev package contains static .a library: optee-os-dev path '/work/imx8mqevk-linaro-linux/optee-os/3.2.0.imx-r0/packages-split/optee-os-dev/usr/include/optee/export-user_ta_arm64/lib/libmbedtls.a'
non -staticdev package contains static .a library: optee-os-dev path '/work/imx8mqevk-linaro-linux/optee-os/3.2.0.imx-r0/packages-split/optee-os-dev/usr/include/optee/export-user_ta_arm64/lib/libutee.a'
non -staticdev package contains static .a library: optee-os-dev path '/work/imx8mqevk-linaro-linux/optee-os/3.2.0.imx-r0/packages-split/optee-os-dev/usr/include/optee/export-user_ta_arm64/lib/libutils.a'
non -staticdev package contains static .a library: optee-os-dev path '/work/imx8mqevk-linaro-linux/optee-os/3.2.0.imx-r0/packages-split/optee-os-dev/usr/include/optee/export-user_ta_arm64/lib/libmpa.a' [staticdev]

Reply:
Those `.a` files ought to be on `FILES_${PN}-staticdev` and a `RDEPENDS_${PN}-dev += "${PN}-staticdev" to be added, so it is broght when doing development when using a SDK.